File: util/crypto_test.go
// Copyright 2023 The Yock Authors. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

package util

import (
	"fmt"
	"testing"
)

func TestCrypto(t *testing.T) {
	Key := `Yock Key`
	Raw := "Hello World!"
	enc := EncodeAESWithKey(Key, Raw)
	dec := DecodeAESWithKey(Key, enc)
	fmt.Printf("Raw: %s\nEnc: %s\nDec: %s\n", Raw, enc, dec)
}

File: lib/test/crypto_test.lua
-- Copyright 2023 The Yock Authors. All rights reserved.
-- Use of this source code is governed by a MIT-style
-- license that can be found in the LICENSE file.

print(crypto.md5("Hello World"))
print(crypto.sha256("Hello World"))
local key = "Yock Key "
local hash = crypto.encode_aes(key, "Hello World!")
print(hash)
print(crypto.decode_aes(key, hash))
